elephant/signal.py

```diff
@@ -0,0 +1,133 @@
+        2. apply a window function to each segment. Hanning window is used by
+            default. This can be changed by giving a window function or an
+            array as parameter *window* (for details, see the documentation of
+            scipy.signal.welch())
+        3. compute the periodogram of each segment
+        4. average the obtained periodograms to yield PSD estimate
+    These steps are implemented in scipy.signal, and this function is a wrapper
+    which provides a proper set of parameters to scipy.signal.welch(). Some
+    parameters for scipy.signal.welch(), such as *nfft*, *detrend*, *window*,
+    *return_onesided* and *scaling*, also works for this function.
+
+    Parameters
+    ----------
+    data: AnalogSignal or AnalogSignalArray
+        time series data, of which PSD is estimated. When an AnalogSignalArray
+        is given, the function computes PSD estimates of single AnalogSignals
+        in the array
+    num_seg: int
+        number of segments. The length of segments is adjusted so that
+        overlapping segments cover the entire stretch of the given data. This
+        parameter is ignored if *len_seg* or *freq_res* is given. Default is 8
+    len_seg: int
+        length of segments. This parameter is ignored if *freq_res* is given.
+        Default is None
+    freq_res: Quantity or float
+        desired frequency resolution of the obtained PSD estimate. Default is
+        None
+    overlap: float
+        overlap between segments represented as a float number between 0 (no
+        overlap) and 1 (complete overlap). Default is 0.5 (half overlapped)
+
+    Returns
+    -------
+    psd: Quantity array
+        estimated PSD. When an AnalogSignalArray is given as input, *psd* is an
+        array of the same shape as the input, which contains PSD estimates of
+        single AnalogSignals in the array
+    freqs Quantity array
+        frequencies associated with the power estimates in *psd*. *freqs* is
+        always a 1-dimensional array irrespective of whether the input is given
+        as AnalogSignal or AnalogSignalArray
+    """
+
+    # initialize a parameter dict (to be given to scipy.signal.welch()) with
+    # the data array
+    params = {'x': np.asarray(data)}
+
+    # if parameters supported by scipy.signal.welch() are given in *kargs*, add
+    # them to the dict. Some of the parameter values may be overridden
+    # afterwards
+    for key, val in kargs.items():
+        if key in ('fs', 'nperseg', 'detrend', 'window', 'noverlap', 'nfft',
+                   'return_onesided', 'scaling'):
+            params[key] = val
+        else:
+            raise ValueError("Unsupported keyword argument '{}'".format(key))
+
+    # specify sampling frequency if the data is given as AnalogSignal(Array)
+    if isinstance(data, neo.AnalogSignal) or \
+            isinstance(data, neo.AnalogSignalArray):
+        params['fs'] = data.sampling_rate.rescale('Hz').magnitude
```

Comment:
OK, I see the point.
Which option would be preferable, either accept NumPy array, list, etc. or raise an exception?